src/IO/NullWriteBuffer.h

```diff
@@ -8,11 +8,14 @@ namespace DB
 {
 
 /// Simply do nothing, can be used to measure amount of written bytes.
-class NullWriteBuffer : public BufferWithOwnMemory<WriteBuffer>, boost::noncopyable
+class NullWriteBuffer : public WriteBuffer, boost::noncopyable
```

Comment:
I remember that we have a butch of WriteBuffers that don't require finalize at all. Like `NullWriteBuffer`, `WriteBufferFromArena`, `MemoryWriteBuffer`, `WriteBufferFromPocoSocket`, etc. and sometimes we use just base class WriteBuffer to read from pointer. I see you added `WriteBufferFromPointer` for this purpose, but AFAIU we still have to call no-op finalize for them (otherwise there will be assert int ~WriteBuffer). In my old PR I tried to solve it by introcuding new base class `WriteBufferWithoutFinalize` that sets `is_finalize_required` flag in `WriteBuffer` to false and used it as base class instead of just `WriteBuffer` when needed. What do you think about it?

Reply:
I think it is good idea. It help to reduce the number of no-op finalize calls.
But I do not address this in that PR. Your proposal I could do in the following changes.